src/core/crypto_tc.c

```diff
@@ -1924,10 +1925,20 @@ int32_t Crypto_TC_ProcessSecurity_Cam(uint8_t *ingest, int *len_ingest, TC_t *tc
         return status; // Cryptography IF call failed, return.
     }
     // Extended PDU processing, if applicable
-    // TODO: Validiate using correct SA
+ 
     if (status == CRYPTO_LIB_SUCCESS && crypto_config.process_sdls_pdus == TC_PROCESS_SDLS_PDUS_TRUE)
     {
-        status = Crypto_Process_Extended_Procedure_Pdu(tc_sdls_processed_frame, ingest);
+        if((sa_ptr->spi == SPI_MIN) || sa_ptr->spi == SPI_MAX) //TODO:  Make sure that SPI can't be changed from the current array index.
```

Comment:
Line 1931: Are we better off to #DEFINE NUM_SA = 64, then in the check assert that sa_ptr->spi >0 and <NUM_SA? Currently we have an off by one case where I feel like we've unintentionally defined 65 SAs here (0 to 64 inclusive). Besides that this logic makes sense.